File: CPP/Film.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

class Film {
private:
  int id;
  string title;
  string genre;
  double rating;
  int release;
  int duration;

public:
  Film(int id, string title, string genre, double rating, int release, int duration) {
    this->id = id;
    this->title = title;
    this->genre = genre;
    this->rating = rating;
    this->release = release;
    this->duration = duration;
  }

  // mengembalikan id film (untuk update dan hapus data film)
  int getFilmId() const { 
    return id; 
  }

  // menampilkan detail data film
  void showFilm() const {
    cout 
    << "ID: " << id 
    << " | Judul: " << title 
    << " | Genre: " << genre 
    << " | Rating: " << rating
    << " | Rilis: " << release
    << " | Durasi: " << duration << " mnt"
    << endl;
  }

  // mengubah data film
  void updateFilm(string newTitle, string newGenre, double newRating, int newRelease, int newDuration) {
    this->title = newTitle;
    this->genre = newGenre;
    this->rating = newRating;
    this->release = newRelease;
    this->duration = newDuration;
  }
};

File: Python/Film.py
class Film:
  def __init__(self, id, title, genre, rating, release, duration):
    self.__id = id
    self.__title = title
    self.__genre = genre
    self.__rating = rating
    self.__release = release
    self.__duration = duration

  # mengembalikan id film (untuk update dan hapus data film)
  def getFilmId(self):
    return self.__id

  # menampilkan detail data film
  def showFilm(self):
    print(f"ID: {self.__id} | Judul: {self.__title} | Genre: {self.__genre} | Rating: {self.__rating} | Rilis: {self.__release} | Durasi: {self.__duration} mnt")

  # mengubah data film
  def updateFilm(self, newTitle, newGenre, newRating, newRelease, newDuration):
    self.__title = newTitle
    self.__genre = newGenre
    self.__rating = newRating
    self.__release = newRelease
    self.__duration = newDuration